src/util/namespace.h

```diff
@@ -84,8 +84,14 @@ class namespacet:public namespace_baset
 
   using namespace_baset::lookup;
 
-  // these do the actual lookup
+
+  /// Searches for a symbol named \p name in `symbol_table1` and returns a
+  /// pointer to it in output parameter \p symbol when found. When no symbol is
+  /// found, it performs the same lookup in `symbol_table2`.
```

Comment:
My audience is someone like me, who 4 days ago didn't know what this method does. I knew a `namespacet` had 2 symbol tables, but didn't know how this method was operating on them.

The intent of my proposed explanation is solving that doubt. I believe that the word "search" is meaningful because only because I'm able to intuitively define what "search" could mean for a symbol table.

Perhaps you mean that I should also explain what "search in a symbol table" is for someone not knowing it? Perhaps I should warn the reader to know first how a `symbol_tablet` works before reading my comment?

Not sure how I could otherwise improve my explanation from your comments.